src/TwillServiceProvider.php

```diff
@@ -478,4 +481,23 @@ public function version()
     {
         return static::VERSION;
     }
+
+    /**
+     * In case 2FA is enabled, we need to check if a QRCode compatible package is
+     * installed.
+     */
+    public function check2FA()
+    {
+        if (!config('twill.enabled.users-2fa')) {
+            return;
+        }
+
+        try {
+            (new User())->get2faQrCode();
```

Comment:
Generating a QR code during every single request when the feature is enabled doesn't sound great to me, am I ready this incorrectly?

Reply:
I do like the idea of letting the developer know like that, but we don't have to avoid the breaking change if we can't, this could wait for Twill 3, right?

Reply:
Horrible actually! I didn't thought it could take too much time, but it's actually too slow, so I just pushed a better implementation.

Timed, running it 100 times:

```
Generating QRCode: 5433ms
Instantiating the class: 1.91ms
Doing nothing: 0.13ms (just commented the line)
```

Can probably be better if I don't instantiate it, but this would mean to make Twill know which classes are necessary for the QRCode to work fine.

Reply:
This is what it does internally: 

<img width="1050" alt="Screenshot 2021-07-09 at 01 06 07" src="https://user-images.githubusercontent.com/3182864/125000957-d8405680-e051-11eb-8c72-16ab40264b55.png">

Reply:
That's great @antonioribeiro! I wonder if we should also scope this check to the console, or to the CMS routes. 1.91ms there is totally fine, but I'd like to avoid unnecessarily affecting the frontend when we're looking for every ms we can get there nowadays.

Reply:
It's actually 0,0191ms/request 😆. But I agree that it's still 2ms and summing to all other ms we have, it's not great.

Moving to check only the existence of the classes (no instantiation) we get down to 0,0082ms/request (1ms).

A call to `App::runningInConsole()` runs in 0,0043ms.

A call to `config('twill.enabled.users-2fa')` runs in 0.0178ms (this one is also execute every time at the beginning and the class check is ran only if 2FA is enabled).

So, yeah, it's not great anyway, and maybe we should just make it a breaking change and only deploy this on the next major. Remembering this part is there just for helping developers not being caught by a broken production because they didn't tested 2FA locally before deploying their apps.

Reply:
A call to `app()->runningInConsole()` runs in 0,0025ms.

A call to `$this->app->runningInConsole()` runs in 0,0016ms. (this is the current implementation)